corehq/tabs/tabclasses.py

```diff
@@ -135,12 +135,20 @@ def _get_tools_items(self):
         from corehq.apps.reports.views import MySavedReportsView
         if isinstance(self.couch_user, AnonymousCouchUser) and PUBLISH_CUSTOM_REPORTS.enabled(self.domain):
             return []
-        return [(_("Tools"), [
-            {'title': _(MySavedReportsView.page_title),
-             'url': reverse(MySavedReportsView.urlname, args=[self.domain]),
-             'icon': 'icon-tasks fa fa-tasks',
-             'show_in_dropdown': True}
-        ])]
+        tools = [{
+            'title': _(MySavedReportsView.page_title),
+            'url': reverse(MySavedReportsView.urlname, args=[self.domain]),
+            'icon': 'icon-tasks fa fa-tasks',
+            'show_in_dropdown': True,
+        }]
+        if toggles.USER_CONFIGURABLE_REPORTS.enabled_for_request(self._request):
```

Comment:
I wouldn't be opposed to adding this link if the toggle is enabled for the user, but I think this is not good for domains where this is enabled. For example rec has this toggle enabled and this would give all users on that project an obvious way to start changing reports and data sources. I don't know their exact project permissions or who has access to their reports, but there's a chance that this exposes the configs to more people than they want. (Of course in an ideal world this page would have more configurable permissions, but there has not been a push/resourcing to do so.

Some background: Unfortunately, the enabling of this flag was not thought through very well in the past and this is an advanced feature that should have as little exposure as possible. This feature flag was originally added with the ability to add domains to it which resulted in more people having access to it than they should. I attempted to make a push to remove domains from the flag before, but it didn't get far since there's reluctance to remove features that people/domains have been given in the past. I've been proactively reaching out to people that have been enabling this for domains as they've been added for the past year or so to at least stop it from getting worse.

Reply:
Will update to just check the user, then - given the current state, that sounds like the best approach.

Reply:
b2705a4cddb